Frameworks/CoreGraphics/CGContext.mm

```diff
@@ -1293,7 +1343,14 @@ static void __CGContextRenderCommandList(CGContextRef context, ID2D1CommandList*
             nullptr);
     }
 
-    deviceContext->DrawImage(commandList);
+    ComPtr<ID2D1Image> currentImage{image};
+
+    ComPtr<ID2D1Effect> shadowEffect;
+    if (__CGContextCreateShadowEffect(context, deviceContext.Get(), currentImage.Get(), &shadowEffect)) {
+        shadowEffect.As(&currentImage);
```

Comment:
I think `FAIL_FAST` is most appropriate here, as we cannot fulfill the user's request at all, but we can't throw an exception since we're behind both an ObjC and a C++ ABI boundary.

FAIL_FAST may not be correct in all cases, and this will require continual evaluation. Alas.